src/EventProducerService.cpp

```diff
@@ -43,11 +43,12 @@ byte EventProducerService::createDefaultEvent(byte evValue)
   unsigned int nodeNum = module_config->nodeNum;
   
   byte index = module_config->findEventSpace();
-  //eventNum = (int)index + 1;
   for (eventNum = 1; eventNum <= module_config->EE_MAX_EVENTS; eventNum++)
   {
     if (module_config->findExistingEvent(nodeNum, eventNum) == module_config->EE_MAX_EVENTS)
     {
+      controller->sendCMDERR(CMDERR_TOO_MANY_EVENTS);
+      controller->sendGRSP(OPC_RQNPN, getServiceID(), CMDERR_INV_PARAM_IDX);
```

Comment:
What happens next? eventNum will be set to EE_MAX_EVENTS. An event is created and overwriting an existing event. Do we want that? 
Yes, I know that it is tricky to pass the error condition back up to the caller chain and eventually block sending an event.

Also the error codes here are triggered by an input event, not the FCU. Who will look at the error code?

Reply:
No.  If the event table is full then we break after sending the error message so nothing should be overwritten.

Reply:
Yes, good point about "break".  What if the event table full returned a value of EE_MAX_EVENTS +1 at line 52?  This could then be caught in the function at line 90 to stop an over-write.

Reply:
Yes, but then that value needs to be handled in findOrCreateEventByEv(). And you need to stop the sendEvent() functions from sending this.

All of this would be easy if we had exceptions. Then you just return back to the main loop and continue with the next action.

Reply:
I agree that this should never happen if the designer seta the EE_MAX_EVENTS to be large enough.  Maybe we park in the "too difficult" file for now.

Reply:
Agreed.